README.md

```diff
@@ -79,57 +84,59 @@ where `ϵ` is the step size of leapfrog integration.
 
 ### Proposal (`proposal`)
 
-- Static HMC with a fixed number of steps (`n_steps`): `StaticTrajectory(int, n_steps)`
-- HMC with a fixed total trajectory length (`len_traj`): `HMCDA(int, len_traj)` 
-- Original NUTS with slice sampling: `NUTS{SliceTS,ClassicNoUTurn}(int)`
-- Generalised NUTS with slice sampling: `NUTS{SliceTS,GeneralisedNoUTurn}(int)`
-- Original NUTS with multinomial sampling: `NUTS{MultinomialTS,ClassicNoUTurn}(int)`
-- Generalised NUTS with multinomial sampling: `NUTS{MultinomialTS,GeneralisedNoUTurn}(int)`
-
-where `int` is the integrator used.
+- Static HMC with a fixed number of steps (`n_steps`): `StaticTrajectory(integrator, n_steps)`
+- HMC with a fixed total trajectory length (`trajectory_length`): `HMCDA(integrator, trajectory_length)` 
+- Original NUTS with slice sampling: `NUTS{SliceTS,ClassicNoUTurn}(integrator)`
+- Generalised NUTS with slice sampling: `NUTS{SliceTS,GeneralisedNoUTurn}(integrator)`
+- Original NUTS with multinomial sampling: `NUTS{MultinomialTS,ClassicNoUTurn}(integrator)`
+- Generalised NUTS with multinomial sampling: `NUTS{MultinomialTS,GeneralisedNoUTurn}(integrator)`
 
 ### Adaptor (`adaptor`)
 
-- Preconditioning on metric space `metric`: `pc = Preconditioner(metric)`
-- Nesterov's dual averaging with target acceptance rate `δ` on integrator `int`: `da = NesterovDualAveraging(δ, int)`
-- Combine the two above *naively*: `NaiveHMCAdaptor(pc, da)`
-- Combine the first two using Stan's windowed adaptation: `StanHMCAdaptor(pc, da)`
+- Adapt the mass matrix of the metric space `metric`: `mma = MassMatrixAdaptor(metric)`
+  - This is lowered to `UnitMassMatrix`, `WelfordVar` or `WelfordCov` based on the type of the metric space.
+- Adapt the step size of the integrator `integrator`: `ssa = StepSizeAdaptor(δ, integrator)`
+  - It uses Nesterov's dual averaging with `δ` as the target acceptance rate.
+- Combine the two above *naively*: `NaiveHMCAdaptor(mma, ssa)`
+- Combine the first two using Stan's windowed adaptation: `StanHMCAdaptor(mma, ssa)`
 
 ### Gradients 
-`AdvancedHMC` supports both AD-based (`Zygote`, `Tracker` and `ForwardDiff`) and user-specified gradients. For the latter, simply replace `ForwardDiff` with `ℓπ_grad` in the ` Hamiltonian`  constructor, where the gradient function `π_grad` should return a tuple containing both the likelihood and its gradient. 
+`AdvancedHMC` supports both AD-based (`Zygote`, `Tracker` and `ForwardDiff`) and user-specified gradients. For the latter, simply replace `ForwardDiff` with `ℓπ_grad` in the `Hamiltonian`  constructor, where the gradient function `π_grad` should return a tuple containing both the likelihood and its gradient. 
```

Comment:
```suggestion
`AdvancedHMC` supports both AD-based (`Zygote`, `Tracker` and `ForwardDiff`) and user-specified gradients. In order to use user-specified gradients, please replace `ForwardDiff` with `ℓπ_grad` in the `Hamiltonian`  constructor, where the gradient function `ℓπ_grad` should return a tuple containing both the log-likelihood and its gradient. 
```

Reply:
@xukai92 For user-specified gradients, should the user define `π_grad`, or ``ℓπ_grad`?

Reply:
It's `ℓπ_grad`. I think we had this typo since long time ago.